packages/better-auth/src/integrations/nestjs/api-error-exception-filter.ts

```diff
@@ -0,0 +1,20 @@
+import type { ArgumentsHost } from "@nestjs/common";
+import { Catch } from "@nestjs/common";
+import type { ExceptionFilter } from "@nestjs/common";
+import { APIError } from "../../api";
+import type { Response } from "express";
+
+@Catch(APIError)
+export class APIErrorExceptionFilter implements ExceptionFilter {
+	catch(exception: APIError, host: ArgumentsHost) {
+		const ctx = host.switchToHttp();
+		const response = ctx.getResponse<Response>();
+		const status = exception.statusCode;
+		const message = exception.body?.message;
+
+		response.status(status).json({
+			statusCode: status,
+			message,
+		});
```

Comment:
Maybe I need to return more properties from the APIError? I don't know if there's a validation errors property for example.